02_bubble/NimishMudhale/bubble_sort_test.go

```diff
@@ -0,0 +1,41 @@
+package main
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestMainOutput(t *testing.T) {
+	// Given
+	r := require.New(t)
+	var buf bytes.Buffer
+	out = &buf
+	// When
+	main()
+	// Then
+	expected := "[1 2 3 5]"
+	actual := buf.String()
+	r.Equalf(expected, actual, "Unexpected output in main()")
+}
+func TestBubbleSortNegativeValues(t *testing.T) {
+	//Given
+	assert := assert.New(t)
+	//when
+	input := []int{-3, -2, -1, -5}
+	sortedArray := bubble(input)
```

Comment:
Nice. Use a few more input/output/expected datasets with [table driven tests](https://github.com/golang/go/wiki/TableDrivenTests), especially corner cases: empty slice, one element slice, slice with duplicated element.
Also ensure that a real copy is made of input.